### PR TITLE
feat: column default value hint/tooltip

### LINKED ISSUE #12665: Table Editor: Custom Function As Default
# Improve documentation

https://app.supabase.com/project/XXX/editor/

Add a link to the page which needs improvement (if relevant)

writing a function in the default field is literal. it doesn't run the function.

to_char(floor(random() * 9000 + 1000), 'FM0000')

Is the documentation missing? Or is it confusing? Why is it confusing?

tell people that if they wrap the function in parenthesis () then it will work

(to_char(floor(random() * 9000 + 1000), 'FM0000'))

A clear and concise description of the improvement.

Add computed default values inside the table editor without the need of running an sql query

Add any other context or screenshots that help clarify your question.

### PR DESCRIPTION
**Pull request by Hallidayo**
## What kind of change does this PR introduce?

Studio > Table Editor

## What is the current behavior?

The hint and tooltip does not mention about using brackets for functions in default values.

## What is the new behavior?

Changed the hint and tooltip to be a bit more clearer:

Screenshot 2023-03-23 at 20 13 54

## Additional context

@gregnr - Does the above need to be mentioned in the docs as well?

Closes #12665

### CODE REVIEW

File: studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/ColumnManagement.tsx [hunk 1]
```diff
@@ -239,7 +239,8 @@ const ColumnManagement: FC<Props> = ({
                     ].join(' ')}
                   >
                     <span className="text-xs text-scale-1200">
-                      Can be either a value or a SQL expression
+                      Can either be a literal or an expression. When using an expression wrap your
+                      expression in brackets e.g (e.g uuid_generate_v4())
```
Comment (saltcod) [inline studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/ColumnManagement.tsx:243]:
```suggestion
                      expression in brackets (e.g uuid_generate_v4())
```

File: studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnDefaultValue.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ const ColumnDefaultValue: FC<Props> = ({
     <InputWithSuggestions
       label="Default Value"
       layout="vertical"
-      description="Can either be a literal or an expression (e.g uuid_generate_v4())"
+      description="Can either be a literal or an expression. When using an expression wrap your expression in brackets e.g (e.g uuid_generate_v4())"
```
Comment (saltcod) [inline studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnDefaultValue.tsx:55]:
```suggestion
      description="Can either be a literal or an expression. When using an expression wrap your expression in brackets (e.g uuid_generate_v4())"
```

**Review by saltcod — APPROVED** [review]

File: studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnDefaultValue.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ const ColumnDefaultValue: FC<Props> = ({
     <InputWithSuggestions
       label="Default Value"
       layout="vertical"
-      description="Can either be a literal or an expression (e.g uuid_generate_v4())"
+      description="Can either be a literal or an expression. When using an expression wrap your expression in brackets (e.g uuid_generate_v4())"
```
Comment (gregnr) [inline studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnDefaultValue.tsx:55]:
```suggestion
      description="Can either be a literal or an expression. When using an expression wrap your expression in brackets, e.g (uuid_generate_v4())"
```

File: studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/ColumnManagement.tsx [hunk 4]
```diff
@@ -239,7 +239,8 @@ const ColumnManagement: FC<Props> = ({
                     ].join(' ')}
                   >
                     <span className="text-xs text-scale-1200">
-                      Can be either a value or a SQL expression
+                      Can either be a literal or an expression. When using an expression wrap your
+                      expression in brackets (e.g uuid_generate_v4())
```
Comment (gregnr) [inline studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/ColumnManagement.tsx:243]:
```suggestion
                      expression in brackets, e.g (uuid_generate_v4())
```

**Review by gregnr — APPROVED** [review]
Thanks for this @Hallidayo and sorry for the delay! Should be sufficient to keep this hint only on studio.

**Pull request merged**